repo-stats/get_data_on_dependencies.py

```diff
@@ -0,0 +1,279 @@
+    new_dict = {}
+    for key in dictionary:
+        new_dict[key.capitalize()] = dictionary[key]
+    return new_dict
+
+
+def get_list_dependencies():
+    #Returns list of dictionaires: [{"version": "", "name": ""}]
+    with open(os.devnull, 'w') as devnull:
+        pip_list = check_output(['pip', 'list', '--format', 'json'],
+                                stderr=devnull, universal_newlines=True)
+        packages_temp = json.loads(pip_list)
+        packages = {}
+        for package in packages_temp:
+            package = capitalize_key_names(package)
+            packages[package["Name"]]=package
+        return packages
+
+
+def get_package_details_str(package_name, try_parsing = True):
+    # runs pip show for given package
+    with open(os.devnull, 'w') as devnull:
+        details = check_output(['pip', 'show', '--verbose', package_name],
+                               stderr=devnull, universal_newlines=True)
+        return details
+
+
+def parse_details_string(detail_string):
+    """pip show --verbose returns a string with details on package
+    string is formated in a sway readable by BytesHeaderParser
+    this function takes a detail string and tries to parse as much data out of it as is possible
+    """
+    final_details = detail_string
+    parsable_details = BytesHeaderParser().parsebytes(final_details.encode())
+    temp_dict = dict(parsable_details.items())
+    if not test_serializability(temp_dict):
+        #something in dict is not serializable, figure it out
+        for key in temp_dict:
+            if not test_serializability({key: temp_dict[key]}):
+                if isinstance(temp_dict[key], email.header.Header):
+                    temp_dict[key] = str(temp_dict[key])
+                else:
+                    raise ValueError("Value not default serializable, please use pdb.set_trace to investigate")
+    try:
+        temp_classifier = parse_classifier(temp_dict["Classifiers"])
+        temp_dict["Classifiers"] = temp_classifier
+    except:
+        temp_dict["Classifiers"] = {"value": temp_dict["Classifiers"]}
+        None 
+    temp_dict["Requires"] = [require.strip() for require in temp_dict["Requires"].split(",")]
+    final_details = temp_dict
+    return final_details
+
+
+def parse_classifier(classifier):
+    """
+    the lines in classifiers in details are structured:
+    value::value::value
+    this function takes classifier straing and tries to read data out of it if possible
+    """
+    lines = classifier.splitlines()
+    splits = []
+    for line in lines:
+        if "::" in line:
+            split = [value.strip() for value in line.split("::")]
+            splits.append(split)
+    return convert_to_dict(splits)
+
+def convert_to_dict(lists):
+    """takes list of lists and tries to compress stackable data 
+    into a dict(works only with one level nesting
+    """
+    if max([len(one_list) for one_list in lists]) == 1:
```

Comment:
The logic in this function seems pretty convoluted to me; is there a simpler approach that could parse out just the data we care about?